### ISSUE TITLE
new CoreAnimation() runtimeType is HtmlElement, unless core-animation.html imported

### ISSUE THREAD

**Issue by davenotik** [dart-lang/core-elements#197]
Take this:

```
var animation = new CoreAnimation();
    print(animation.runtimeType);
```
That prints `HtmlElement` unless I have the `core-animation.html` imported in the template:

`<link rel="import" href="../../../../../../../packages/core_elements/core_animation.html">`

**Comment by jakemac53** [comment]
This is actually expected behavior unfortunately, although with some new changes coming down the pipeline it will no longer be an issue (more info to come with the 0.16.0 release).

The constructor for CoreAnimation actually just looks like `document.createElement('core-animation')`. So if you haven't registered that tag with the document yet then it will not actually be a CoreAnimation object. Under checked mode this probably also throws an error, although I haven't confirmed.

Also, its good to note that the custom elements polyfill is actually async, so in other browsers it will definitely not be upgraded until the end of the current microtask.

**Comment by davenotik** [comment]
Gotcha. To clarify, it's cool that I just have the core-animation.html imported (i.e. the element is registered) even without adding any `<core-animation>` to the template (my code adds that at runtime in my case).

**Comment by jakemac53** [comment]
Yes, as it stands today you have to include the html import even if you don't use it in the main template.

**Comment by jakemac53** [comment]
This is fixed in polymer 0.16.0 thanks to @HtmlImport, yay!
